File: baekjoon/자료 구조/[10828]스택.py
import sys

n = int(input())

stack = []

for i in range(n):
    command = sys.stdin.readline().split()
    
    if command[0] == "push":
        stack.append(command[1])
    if command[0] == "pop":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    if command[0] == "size":
        print(len(stack))
    if command[0] == "empty":
        if len(stack) == 0:
            print(1)
        else: 
            print(0)
    if command[0] == "top":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

File: baekjoon/자료 구조/[10828]스택.cpp
#include <iostream>
#include <algorithm>
#include <stack>
#include <string>
using namespace std;

int main() {
    int n;
    string com;
    stack <int> st;

    cin >> n;
    
    for (int i = 0; i < n; i++) {
        cin >> com;

        if (com == "push") {
            int temp;
            cin >> temp;
            st.push(temp);
        }
        else if (com == "pop") {
            if (!st.empty()) {
                cout << st.top() << "\n";
                st.pop();
            }
            else cout << -1 << "\n";
        }
        else if (com == "size") {
            cout << st.size() << "\n";
        }
        else if (com == "empty") {
            cout << st.empty() << "\n";
        }
        else if (com == "top") {
            if (!st.empty()) cout << st.top() << "\n";
            else cout << -1 << "\n";            
        }

    }

    return 0;
}